src/runtime-rs/crates/resource/src/volume/spdk_volume.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2023 Alibaba Cloud
+// Copyright (c) 2023 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use nix::sys::{stat, stat::SFlag};
+use tokio::sync::RwLock;
+
+use super::Volume;
+use crate::volume::utils::{
+    generate_shared_path, volume_mount_info, DEFAULT_VOLUME_FS_TYPE, KATA_SPDK_VOLUME_TYPE,
+    KATA_SPOOL_VOLUME_TYPE,
+};
+use hypervisor::{
+    device::{
+        device_manager::{do_handle_device, DeviceManager},
+        DeviceConfig, DeviceType,
+    },
+    VhostUserConfig, VhostUserType,
+};
+
+#[derive(Clone)]
+pub(crate) struct SPDKVolume {
+    storage: Option<agent::Storage>,
+    mount: oci::Mount,
+    device_id: String,
+}
+
+/// SPDKVolume: spdk block device volume
```

Comment:
Maybe this comment written in l24 is better?

Reply:
fixed